### ISSUE TITLE
Possible bug in Socket.sendAll function

### ISSUE THREAD

**Issue by pblasucci** [zeromq/fszmq#121]
Currently, if you send an empty container (`seq`, `list`, `array`, et cetera) to the `Socket.recvAll` function you get the error

```
﻿﻿﻿System.IndexOutOfRangeException: Array index is out of range.
  at fszmq.SocketModule.SendAll (fszmq.Socket socket, IEnumerable`1 message) [0x000cb] in ... /fszmq/src/fszmq/Socket.fs:136 
```
, which is caused by passing a (computed) negative value to `Seq.take`. 

~~Is this the desired behavior?~~ No. As a convenience, an empty container should be treated as a container with a single empty frame.

**Comment by coconaut** [comment]
I'd love to give this one a try.

**Comment by pblasucci** [comment]
@coconaut Awesome! Have at it. While you're in the code, there's an additional micro-optimization that'd be nice. Specifically, if the input only has a single item, we can just directly call `send` (thereby skipping a bunch of processing).

If you have any questions, please don't hesitate to ask. Oh, also, just as an FYI, you may want to read [the developer notes](https://github.com/zeromq/fszmq/blob/master/CONTRIBUTING.md), as the spell out a few subtleties in the code base.

Cheers!

**Comment by coconaut** [comment]
Thanks! Ah, that makes sense. I see what you're talking about and will definitely work it in. 

It took me a bit to get my testing up and running (really got jammed by forgetting a thread.sleep) but I think I've got a handle on it.

**Comment by pblasucci** [comment]
Fixed via #123
